lib/astc_codec.cpp

```diff
@@ -291,38 +293,242 @@ astcVkFormat(ktx_uint32_t block_size, bool sRGB) {
 
 /**
  * @memberof ktxTexture
- * @ingroup writer
+ * @internal
+ * @ingroup reader
  * @~English
- * @brief Creates valid ASTC encoder action from string.
+ * @brief       Should be used to get uncompressed version of ASTC VkFormat
  *
- * @return      Valid astc_profile from string
+ * The decompressed format is calculated from coressponding ASTC format. There are
+ * only 3 possible options currently supported. RGBA8, sRGBA8 and RGBA32.
+ *
+ * @return      Uncompressed version of VKFormat for a specific ASTC VkFormat
  */
-static astcenc_profile
-astcEncoderAction(const ktxAstcParams &params, const uint32_t* bdb) {
+inline VkFormat getUncompressedFormat(VkFormat format) noexcept {
+    switch (format) {
+    case VK_FORMAT_ASTC_4x4_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_5x4_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_5x5_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_6x5_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_6x6_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_8x5_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_8x6_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_8x8_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_10x5_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_10x6_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_10x8_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_10x10_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_12x10_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_12x12_UNORM_BLOCK:
+    case VK_FORMAT_ASTC_3x3x3_UNORM_BLOCK_EXT:
+    case VK_FORMAT_ASTC_4x3x3_UNORM_BLOCK_EXT:
+    case VK_FORMAT_ASTC_4x4x3_UNORM_BLOCK_EXT:
+    case VK_FORMAT_ASTC_4x4x4_UNORM_BLOCK_EXT:
+    case VK_FORMAT_ASTC_5x4x4_UNORM_BLOCK_EXT:
+    case VK_FORMAT_ASTC_5x5x4_UNORM_BLOCK_EXT:
+    case VK_FORMAT_ASTC_5x5x5_UNORM_BLOCK_EXT:
+    case VK_FORMAT_ASTC_6x5x5_UNORM_BLOCK_EXT:
+    case VK_FORMAT_ASTC_6x6x5_UNORM_BLOCK_EXT:
+    case VK_FORMAT_ASTC_6x6x6_UNORM_BLOCK_EXT:
+        return VK_FORMAT_R8G8B8A8_UNORM;
```

Comment:
This is not a 3d texture format. If the incoming texture is actually 3d what does DecodeAstc do, decode only depth 0? Fail? Perhaps we should return KTX_UNSUPPORTED_FEATURE for 3d textures for now.

Reply:
Its not any format, 1D, 2D or 3D. It has no dimensionality in it, its just a format.
For 3D textures it doesn't do anything fancy. Just decompresses each slice into a separate 2D texture and sets the correct `This->baseDepth`. In the metrics location we are comparing each slice separately as 2D textures.

Maybe the function should be renamed to `getUncompressedFormatPerSlice` but thats too descriptive I think. Its functionality looks correct to me.